### PR TITLE
Add support for '%' type to output floating point values as a percentage.

### PR DESCRIPTION
**Pull request by gawain-bolton**
This helps with compatibility with Python's format strings.

### CODE REVIEW

**Comment by vitaut** [comment]
Thanks for the PR, but please fix the CI build, apply clang-format to your changes and add a unit test.

**Comment by gawain-bolton** [comment]
> Thanks for the PR, but please fix the CI build, apply clang-format to your changes and add a unit test.

I have applied clang-format and added unit tests.
The CI builds also now pass.

File: include/fmt/format.h [hunk 1]
```diff
@@ -2854,26 +2867,39 @@ void basic_writer<Range>::write_double(T value, const format_specs& spec) {
...
-  if (internal::fputil::isinfinity(value))
-    return write_inf_or_nan(handler.upper ? "INF" : "inf");
+  if (internal::fputil::isnotanumber(value)) {
+    write_inf_or_nan(handler.upper ? "NAN" : "nan");
+    return;
+  }
+  if (internal::fputil::isinfinity(value)) {
+    write_inf_or_nan(handler.upper ? "INF" : "inf");
+    return;
+  }
```
Comment (vitaut) [inline include/fmt/format.h]: This change looks unrelated. Please revert.

File: include/fmt/format.h [hunk 2]
```diff
@@ -2854,26 +2867,39 @@ void basic_writer<Range>::write_double(T value, const format_specs& spec) {
...
+
   bool use_grisu = fmt::internal::use_grisu<T>() &&
                    (!spec.type || handler.fixed) &&
                    internal::grisu2_format(static_cast<double>(value), buffer,
                                            precision, handler.fixed, exp);
   if (!use_grisu) internal::sprintf_format(value, buffer, spec);
+
+  if (handler.as_percentage) {
+    buffer.push_back('%');
+    --exp;  // Adjust decimal place position.
```
Comment (vitaut) [inline include/fmt/format.h]: Shouldn't this be -= 2?

File: include/fmt/format.h (hunk 1 above)
Comment (gawain-bolton) [reply include/fmt/format.h]: You're right, I've reverted this changes.

File: include/fmt/format.h (hunk 2 above)
Comment (gawain-bolton) [reply include/fmt/format.h]: The decimal place position is shifted by one place because only one character, the percent sign, is added.

**Comment by vitaut** [comment]
Merged in https://github.com/fmtlib/fmt/commit/79b79f329ea126c14d150d9b75668ed137b33c42, thanks!

**Comment by vitaut** [comment]
@gawain-bolton, could you please review the updated [CONTIBUTING](https://github.com/fmtlib/fmt/blob/ebec00138bd10e949ff34955bcb096387cd32acc/CONTRIBUTING.md) document, particularly the part about licensing, and let me know if you agree with it being applied to your contributions to {fmt}? The library is likely to be relicensed (#1073) so I'm collecting approval from all earlier contributors. Thanks!
